### PR TITLE
[Swift] Make classes and members of classes 'open' instead of 'public' [4.0]

### PR DESCRIPTION
**Pull request by slavapestov**
Fixes <rdar://problem/29039602>.

### CODE REVIEW

**Comment by slavapestov** [comment]
@jrose-apple Is this what you had in mind?

File: source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp [hunk 1]
```diff
@@ -835,10 +835,16 @@ void SwiftASTManipulator::MakeDeclarationsPublic() {
     virtual bool walkToDeclPre(swift::Decl *decl) {
       if (swift::ValueDecl *value_decl =
               llvm::dyn_cast<swift::ValueDecl>(decl)) {
-        value_decl->overwriteAccessibility(swift::Accessibility::Public);
+        auto access = swift::Accessibility::Public;
+        if (swift::isa<swift::ClassDecl>(value_decl) ||
+            swift::isa<swift::ClassDecl>(value_decl->getDeclContext())) {
```
Comment (jrose-apple) [inline source/Plugins/ExpressionParser/Swift/SwiftASTManipulator.cpp:840]: Nitpick: `llvm::isa`. (We just borrow it into our namespace.)

**Review by jrose-apple — COMMENTED** [review]
Yep, strictly an improvement over what we have today. Still would want to figure out what to do with extension methods (`@objc` ones can be overridden), but this is good enough and won't put complicated logic into LLDB.

**Comment by jrose-apple** [comment]
Needs a test case, of course.

**Comment by jimingham** [comment]
This looks fine. The test case repl/subclassing should test exactly this case.  The commands typed in by hand clearly fail.

**Review by jimingham — COMMENTED** [review]
This looks fine except we have to figure out why the repl/subclassing test didn't catch the failure.  When you type its commands in by hand it clearly fails.

**Comment by jrose-apple** [comment]
Looks like that test is XFAILed for unrelated reasons: https://github.com/apple/swift-lldb/blob/stable/packages/Python/lldbsuite/test/repl/subclassing/TestREPLSubclassing.py#L28

**Comment by jimingham** [comment]
Grr...  Yeah, these tests were originally written using pexpect, but pexpect turned out to be weirdly flakey.  We need to convert all these tests to use the expression parser in repl mode.

**Comment by jimingham** [comment]
I take that back, looks like Enrico did make these REPLTests use the expression parser in REPL mode rather than pexpect.  The x-fail was just for flakiness of the pexpect mechanism, so it looks like the missing step was to un-xfail this test.

**Comment by jimingham** [comment]
To get the record straight, it was Sean that did the conversion...

**Comment by slavapestov** [comment]
@jimingham If I re-enable the test, it does not fail. It looks like dotest.py doesn't run REPL tests at all?

**Comment by slavapestov** [comment]
It looks like dotest runs a test if its defined like so:

```class REPLSubclassingTestCase (lldbrepl.REPLTest)```

But not like so

```class REPLSubclassingTestCase (REPLTest)```

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
Ok, here is a new version with tests. @jimingham and @jrose-apple does this look OK for 4.0?

**Comment by slavapestov** [comment]
Also I ran the tests locally but I don't know how to trigger CI here...

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test

**Review by jimingham — APPROVED** [review]
I don't have any objections, but I defer to Jordan for whether "open" is the best way to mark classes in the current REPL, his knowledge of the swift visibility model far exceeds mine.

**Comment by jrose-apple** [comment]
Slava came up with a few other cases that still won't work, but they're not regressions. This seems like a strict improvement to me, so I'm 👍 as well.

**Comment by jimingham** [comment]
Slava, can you file bugs on the cases that don't work?  Maybe we need to rethink how we accumulate code in the REPL to get this all to work smoothly.

**Pull request merged**